lib/rucio/core/rule.py

```diff
@@ -944,9 +1094,14 @@ def delete_rule(rule_id, purge_replicas=None, soft=False, delete_parent=False, n
             insert_rule_history(rule=rule, recent=True, longterm=False, session=session)
             return
 
-        locks = session.query(models.ReplicaLock)\
-                       .filter(models.ReplicaLock.rule_id == rule_id)\
-                       .with_for_update(nowait=nowait).yield_per(100)
+        stmt = select(
+            models.ReplicaLock
+        ).with_for_update(
+            nowait=nowait
+        ).where(
+            models.ReplicaLock.rule_id == rule_id
+        )
+        locks = session.execute(stmt).yield_per(100).scalars()
```

Comment:
Again, is this correct to do the .scalars() at this point?

Reply:
Scalars are not needed here either, removed it. As mentioned, I will figure out how yield_per().scalars() is handled for future use cases.